### PR TITLE
Python mfa

### PR DESCRIPTION
**Pull request by sasakalaba**
Do not merge.

### CODE REVIEW

**Comment by rdegges** [comment]
This looks good so far! It sounds like updating the code snippets is the remaining bit of work to be done.

**Comment by jakubvul** [comment]
@rdegges please advise.

File: source/code/python/authentication/mfa_add_ga_factor_req.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+google_factor = account.factors.create({
+    'type': 'google-authenticator',
+    'accountName': 'example@stormpath.com',
+    'issuer': 'myEnterpriseIssuer'}
+)
```
Comment (rdegges) [inline source/code/python/authentication/mfa_add_ga_factor_req.py]: Can we get the closing curly brace on the bottom line here? This way it looks nicer.

File: source/code/python/authentication/mfa_add_sms_factor_req.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+sms_factor = account.factors.create({
+    'phone': {'number': '+12223334444'},
+    'type': 'SMS'},
```
Comment (rdegges) [inline source/code/python/authentication/mfa_add_sms_factor_req.py]: Can we also get the closing curly brace here moved down? This way it'll look a bit nicer, eg:

```python
sms_factor = account.factors.create({
    'phone': {'number': '+12223334444'},
    'type': 'SMS',
}, challenge=False)

File: source/code/python/authentication/mfa_challenge_sms_factor_req.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Setting a message is optional, set a custom message only if you want to
+# overwrite the default message. Make sure you use a "${code}" placeholder
+# in your custom message.
+challenge = sms_factor.challenge_factor(
+    message='For the sake of example, your code is ${code}')
```
Comment (rdegges) [inline source/code/python/authentication/mfa_challenge_sms_factor_req.py]: Let's get this on one line as well. It'll render nicer that way in the Sphinx markup, and be easier to copy+paste.

File: source/code/python/authentication/mfa_create_and_challenge_default.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+sms_factor = account.factors.create({
+    'phone': {'number': '+12223334444'},
+    'type': 'SMS'},
+    challenge=True)
```
Comment (rdegges) [inline source/code/python/authentication/mfa_create_and_challenge_default.py]: Same formatting issue as above.

File: source/code/python/authentication/mfa_create_and_challenge_message.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+sms_factor = account.factors.create({
+    'phone': {'number': '+12223334444'},
+    'challenge': {'message': 'For the sake of example, your code is ${code}'},
+    'type': 'SMS'},
+    challenge=True
```
Comment (rdegges) [inline source/code/python/authentication/mfa_create_and_challenge_message.py]: Same formatting issue as above.

**Comment by rdegges** [comment]
Hey @sasakalaba, just left a few small style update comments to this PR. Would you get those in? Then we can merge!

Looks really good!

**Comment by sasakalaba** [comment]
@rdegges code snippets updated.

**Comment by rdegges** [comment]
Fantastic, thanks so much @sasakalaba!

**Pull request merged**